File: Data Structures & Algorithms/house-robber/submission-1.cpp
class Solution {
public:
int func(int i,vector<int> &nums,vector<int> &dp){
    if(i>=nums.size()){
        return 0;
    }
    if(dp[i]!=-1) return dp[i];
     return dp[i]=max(nums[i]+func(i+2,nums,dp),func(i+1,nums,dp));
}
    int rob(vector<int>& nums) {
        int n=nums.size();
        vector<int> dp(n+1,-1);
        return func(0,nums,dp);
        
    }
};

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-1.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        my_set=set()
        maxL=0
        l=0
        r=1
        for r in range(len(s)):
            while s[r] in my_set:
                my_set.remove(s[l])
                l+=1    
            my_set.add(s[r])
            maxL=max(maxL,r-l+1)
        return maxL


       
                
        